hcipy/aperture/generic.py

```diff
@@ -302,13 +320,27 @@ def make_spider(p1, p2, spider_width):
 	spider = rectangular_aperture((spider_length, spider_width))
 
 	def func(grid):
-		x, y = grid.shifted(-shift).coords
+		g = grid.as_('cartesian')
+
+		if g.is_separated:
+			x, y = g.separated_coords
+			x = x[np.newaxis, :]
+			y = y[:, np.newaxis]
+		else:
+			x, y = g.coords
+
+		x = x - shift[0]
+		y = y - shift[1]
 
 		x_new = x * np.cos(spider_angle) + y * np.sin(spider_angle)
 		y_new = y * np.cos(spider_angle) - x * np.sin(spider_angle)
-		spider_grid = CartesianGrid(UnstructuredCoords([x_new, y_new]))
 
-		return 1 - spider(spider_grid)
+		spider = x_new <= (spider_length / 2)
```

Comment:
Wouldn't 
```
spider = abs(x_new) < spider_length / 2
spider *= abs(y_new) < spider_width / 2
```
be cleaner?

Reply:
See https://github.com/ehpor/hcipy/pull/140#issuecomment-1219615161.